_studio/shared/src/mfx_vpp_vaapi.cpp

```diff
@@ -1231,35 +1231,37 @@ mfxStatus VAAPIVideoProcessing::Execute(mfxExecuteParams *pParams)
     if((pParams->bEOS) && (pParams->bDeinterlace30i60p == true))
         m_deintFrameCount = 0;
 
-#if defined(LINUX_TARGET_PLATFORM_BXTMIN) || defined(LINUX_TARGET_PLATFORM_BXT)
-// It looks like only BXT supports this at the moment
-#define VPP_NO_COLORFILL
-#endif
+    bool VppNoColorfill =
+            // It looks like only BXT supports this at the moment
+            (MFX_HW_APL == hwType)
+            // Do not disable colorfill if letterboxing is used
+            && (0 == output_region.x && 0 == output_region.y);
```

Comment:
@alariono :
* Why w/h is not checked here?
* Do we somehow program a colorfill color at the first place? if yes, do we expose user API for this?

Reply:
@dvrogozh 
>Why w/h is not checked here?
Because this is not need for letterboxing filter detection.
>Do we somehow program a colorfill color at the first place? if yes, do we expose user API for this?
No, except for VPP composition. MSDK have hard cored value for background.
See link https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/src/mfx_vpp_vaapi.cpp#L1057

Reply:
@dvrogozh 
>Why w/h is not checked here?

Because this is not need for letterboxing filter detection.

>Do we somehow program a colorfill color at the first place? if yes, do we expose user API for this?

No, except for VPP composition. MSDK have hard cored value for background.
See link https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/src/mfx_vpp_vaapi.cpp#L1057

Reply:
I don't like how no color fill support is implemented. And I still think that condition is wrong. Yes, we will support letter boxing with the color fill, but why we permanently disable it for other cases? What if user wants to render some picture in the middle of the surface and have everything else filled with red?

Finally, I see this change in media-driver: https://github.com/intel/media-driver/pull/324. If I understands correctly, it disables color fill if color value is 0. This seems to be much more reasonable solution than what we are trying to achieve with this patch.

I don't approve this patch and vote for the try out of the driver's PR I have mentioned + make similar changes in mediasdk if it works.